### PR TITLE
Prep for jenkins

### PR DESCRIPTION
**Pull request by maratoid**
Add option to use docker without docker-machine in .sh scripts
split up .tfvars per cluster name to avoid resource naming conflicts.

### CODE REVIEW

**Comment by maratoid** [comment]
PLEASE DO NOT MERGE YET

File: bin/kraken-up.sh [hunk 1]
```diff
@@ -44,17 +44,19 @@ if [ ! -f "${KRAKEN_ROOT}/terraform/${KRAKEN_CLUSTER_TYPE}/Dockerfile" ]; then
...
+if [ "${KRAKEN_NATIVE_DOCKER}" = false ]; then 
+  if docker-machine ls -q | grep --silent "${KRAKEN_DOCKER_MACHINE_NAME}"; then
+    inf "Machine ${KRAKEN_DOCKER_MACHINE_NAME} already exists."
+  else
+    if [ -z ${KRAKEN_DOCKER_MACHINE_OPTIONS+x} ]; then
+      error "--dmopts not specified. Docker Machine option string is required unless machine ${KRAKEN_DOCKER_MACHINE_NAME} already exists."
+      exit 1
+    fi
+    setup_dockermachine
   fi
```
Comment (spiffxp) [inline bin/kraken-up.sh]: Seems like the docker machine logic is copy pasted in all the `bin/kraken-*.sh` files, maybe consolidate in `utils.sh` since that's already being sourced?

File: bin/kraken-down.sh [hunk 2]
```diff
@@ -47,11 +52,20 @@ if ! docker inspect kraken_data &> /dev/null; then
...
-docker run -d --name kraken_cluster --volumes-from kraken_data \
+inf "Tearing down kraken cluster:\n  'docker run -d --name ${kraken_container_name} --volumes-from kraken_data samsung_ag/kraken terraform destroy -force -input=false -state=/kraken_data/${KRAKEN_CLUSTER_NAME}/terraform.tfstate /opt/kraken/terraform/${KRAKEN_CLUSTER_TYPE}'"
+docker run -d --name ${kraken_container_name} --volumes-from kraken_data \
   samsung_ag/kraken bash -c \
-  "until terraform destroy -force -input=false -var-file=/opt/kraken/terraform/${KRAKEN_CLUSTER_TYPE}/terraform.tfvars \
-    -state=/kraken_data/${KRAKEN_CLUSTER_NAME}/terraform.tfstate /opt/kraken/terraform/${KRAKEN_CLUSTER_TYPE}; do echo 'Retrying...'; sleep 5; done"
+  "until terraform destroy -force -input=false -var-file=/opt/kraken/terraform/${KRAKEN_CLUSTER_TYPE}/${KRAKEN_CLUSTER_NAME}/terraform.tfvars \
+    -state=/kraken_data/${KRAKEN_CLUSTER_NAME}/terraform.tfstate /opt/kraken/terraform/${KRAKEN_CLUSTER_TYPE}; do echo 'Retrying...'; sleep 5; done;"
```
Comment (spiffxp) [inline bin/kraken-down.sh]: Rather than a big one-liner, can we extract this into a script that's added into the docker container?

File: bin/kraken-up.ps1 [hunk 3]
```diff
@@ -60,26 +60,27 @@ If ($LASTEXITCODE -eq 0) {
...
+  inf "Removing old kraken_cluster container:`n   'docker rm -f $kraken_container_name'"
+  Invoke-Expression "docker rm -f $kraken_container_name"
 }
 
 inf "Building kraken container:`n  'docker build -t samsung_ag/kraken -f '$krakenRoot/terraform/$clustertype/Dockerfile' '$krakenRoot' '"
 Invoke-Expression "docker build -t samsung_ag/kraken -f '$krakenRoot/terraform/$clustertype/Dockerfile' '$krakenRoot'"
 
 # run cluster up
-$command =  "docker run -d --name kraken_cluster --volumes-from kraken_data samsung_ag/kraken bash -c " + 
+$command =  "docker run -d --name $kraken_container_name --volumes-from kraken_data samsung_ag/kraken bash -c " +
```
Comment (spiffxp) [inline bin/kraken-up.ps1]: Rather than a big one-liner, can we extract this into a script that's added into the docker container?

File: bin/kraken-up.sh [hunk 4]
```diff
@@ -81,14 +84,14 @@ inf "Building kraken container:\n  'docker build -t samsung_ag/kraken -f \"${KRA
 docker build -t samsung_ag/kraken -f "${KRAKEN_ROOT}/terraform/${KRAKEN_CLUSTER_TYPE}/Dockerfile" "${KRAKEN_ROOT}"
 
 # run cluster up
-inf "Building kraken cluster:\n  'docker run -d --volumes-from kraken_data samsung_ag/kraken terraform apply \
-  -input=false -state=/kraken_data/${KRAKEN_CLUSTER_NAME}/terraform.tfstate -var-file=/opt/kraken/terraform/${KRAKEN_CLUSTER_TYPE}/terraform.tfvars /opt/kraken/terraform/${KRAKEN_CLUSTER_TYPE}'"
-docker run -d --name kraken_cluster --volumes-from kraken_data samsung_ag/kraken bash -c "\
+inf "Building kraken cluster:\n  'docker run -d --name ${kraken_container_name} --volumes-from kraken_data samsung_ag/kraken terraform apply \
+  -input=false -state=/kraken_data/${KRAKEN_CLUSTER_NAME}/terraform.tfstate -var-file=/opt/kraken/terraform/${KRAKEN_CLUSTER_TYPE}/${KRAKEN_CLUSTER_NAME}/terraform.tfvars /opt/kraken/terraform/${KRAKEN_CLUSTER_TYPE}'"
+docker run -d --name ${kraken_container_name} --volumes-from kraken_data samsung_ag/kraken bash -c "\
```
Comment (spiffxp) [inline bin/kraken-up.sh]: Rather than a big one-liner, can we extract this into a script that's added into the docker container?  Maybe this saves us some dupe between the .sh and .ps1 variants

File: bin/kraken-ssh.sh [hunk 5]
```diff
@@ -23,18 +25,27 @@ if [ -z ${KRAKEN_CLUSTER_NAME+x} ]; then
...
+  error "Cluster build is currently running:\n Run\n  'docker logs --follow ${kraken_container_name}'\n to see logs."
   exit 1
 fi
-eval "$(docker-machine env ${KRAKEN_DOCKER_MACHINE_NAME})"
 
 mkdir -p "clusters/${KRAKEN_DOCKER_MACHINE_NAME}/${KRAKEN_CLUSTER_NAME}"
 docker cp kraken_data:/kraken_data/${KRAKEN_CLUSTER_NAME}/ssh_config "clusters/${KRAKEN_DOCKER_MACHINE_NAME}/${KRAKEN_CLUSTER_NAME}/ssh_config"
-docker cp kraken_cluster:/root/.ssh/id_rsa "clusters/${KRAKEN_DOCKER_MACHINE_NAME}/id_rsa"
-docker cp kraken_cluster:/root/.ssh/id_rsa.pub "clusters/${KRAKEN_DOCKER_MACHINE_NAME}/id_rsa.pub"
+docker cp ${kraken_container_name}:/root/.ssh/id_rsa "clusters/${KRAKEN_DOCKER_MACHINE_NAME}/id_rsa"
```
Comment (spiffxp) [inline bin/kraken-ssh.sh]: Should this be clusters/${KRAKEN_DOCKER_MACHINE_NAME}/${KRAKEN_CLUSTER_NAME}/id_rsa?

Also, why do we need the kraken docker machine name as an index into the clusters dir? Implies there's something meaningful about using two different docker-machine instances to stand up the same cluster name?

File: bin/kraken-ssh.sh [hunk 6]
```diff
@@ -23,18 +25,27 @@ if [ -z ${KRAKEN_CLUSTER_NAME+x} ]; then
...
+
+kraken_container_name="kraken_cluster_${KRAKEN_CLUSTER_NAME}"
+is_running=$(docker inspect -f '{{ .State.Running }}' ${kraken_container_name})
+if [ ${is_running} == "true" ];  then
+  error "Cluster build is currently running:\n Run\n  'docker logs --follow ${kraken_container_name}'\n to see logs."
   exit 1
 fi
-eval "$(docker-machine env ${KRAKEN_DOCKER_MACHINE_NAME})"
 
 mkdir -p "clusters/${KRAKEN_DOCKER_MACHINE_NAME}/${KRAKEN_CLUSTER_NAME}"
```
Comment (spiffxp) [inline bin/kraken-ssh.sh]: It would be nice if `clusters/` was explicitly specified relative to KRAKEN_ROOT, looks like this will dump into the current working directory otherwise?

**Comment by spiffxp** [comment]
retest this please

**Comment by spiffxp** [comment]
retest this please

**Comment by spiffxp** [comment]
retest this please

File: bin/kraken-up.ps1 (hunk 3 above)
Comment (maratoid) [inline bin/kraken-up.ps1]: I'd prefer to do this in a separate PR

File: bin/kraken-up.sh (hunk 4 above)
Comment (maratoid) [inline bin/kraken-up.sh]: I'd prefer to do this in a separate PR

**Comment by maratoid** [comment]
Ready to be reviewed

File: bin/kraken-connect.sh [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+fi
+
+kraken_container_name="kraken_cluster_${KRAKEN_CLUSTER_NAME}"
+is_running=$(docker inspect -f '{{ .State.Running }}' ${kraken_container_name})
+if [ ${is_running} == "true" ];  then
+  error "Cluster build is currently running:\n Run\n  'docker logs --follow ${kraken_container_name}'\n to see logs."
+  exit 1
+fi
+
+mkdir -p "clusters/${KRAKEN_DOCKER_MACHINE_NAME}/${KRAKEN_CLUSTER_NAME}"
```
Comment (spiffxp) [inline bin/kraken-connect.sh]: Can we make this relative to KRAKEN_ROOT rather than relative to cwd

**Comment by spiffxp** [comment]
retest this please

**Comment by spiffxp** [comment]
retest this please

**Comment by spiffxp** [comment]
retest this please

**Comment by spiffxp** [comment]
retest this please

**Comment by spiffxp** [comment]
retest this please
